### DISCUSSION TITLE
issue: I try to upload file using react hook form

### DISCUSSION THREAD

**Discussion by siamahnaf** [react-hook-form/react-hook-form#6743] [Q&A] [answered]
### Version Number

7

### Codesandbox/Expo snack

https://codesandbox.io/s/youthful-glade-d4kq8

### Expected behaviour

Here is my code-
import React from "react";
import {useForm} from "react-hook-form";

function Test() {
    const {register, handleSubmit} = useForm();
    const onSubmit = (data) => {
        console.log(data);
    };
    return (
        <form onSubmit={handleSubmit(onSubmit)}>
            <input {...register} type="file" name="picture"/>
            <button>Submit</button>
        </form>
    );
}

### What browsers are you seeing the problem on?

Firefox, Chrome, Edge

### Relevant log output

```shell
I see in log output an empty object.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fahadsohail482** [comment, marked as answer]
@siamahnaf198 Please provide the name within the register function like so. 

> register("anyName") 

your input should look like this and remove the name attribute from your input.
`<input {...register("picture")} type="file" />`

[Here is the working example](https://codesandbox.io/s/serene-leaf-kx9l8?file=/pages/index.js)

**Reply by siamahnaf** [reply]
Hello, Thanks for your help. I have facing one another problem. When I use e.target.reset() in onSubmit function then I can't find file. If I remove it, I successfully find file in console. How can I solve it.

Here is Sandbox link-
https://codesandbox.io/s/icy-smoke-m7gcz

From codesandbox: I have to use e.target.reset for reseting name filed.

**Reply by fahadsohail482** [reply]
@siamahnaf198 It's recommended to not invoke reset inside onReset or onSubmit callback.
Please follow this [sandbox](https://codesandbox.io/s/react-hook-form-handlesubmit-with-reset-xu1zu) on how to reset the fields.
https://react-hook-form.com/api/useform/reset/

**Reply by siamahnaf** [reply]
I can't find any solution.

I do same to your sandbox. But there is thrown a error message like-

TypeError: HTMLInputElement.files setter: Value being assigned does not implement interface FileList.

**Reply by siamahnaf** [reply]
Please solve to my  Sandbox code

**Reply by siamahnaf** [reply]
Hello, I can't solve anytihing, Please help

**Reply by fahadsohail482** [reply]
@siamahnaf198 here is the updated sandbox link.
[Link](https://codesandbox.io/s/busy-poitras-xsmpw?file=/pages/index.js:409-539)

**Reply by siamahnaf** [reply]
Please see. In your updated sandbox link there no file data after submitted in console. You can see it.

**Reply by fahadsohail482** [reply]
Do you want to keep the file once you submit it? Please provide me with more information.

**Reply by siamahnaf** [reply]
When I submit it, it remove the file. But in onSubmit data, I need file to sent backend. But your codeSandBox link after submit there is no file in submited data.

**Reply by fahadsohail482** [reply]
oh, I have added comments in code sandbox what to do.
[Link](https://codesandbox.io/s/zealous-sound-bwmud?file=/pages/index.js)

**Comment by bluebill1049** [comment]
thanks @fahadsohail482 👍

**Reply by siamahnaf** [reply]
I still face problem when trying to reset() data. Please help

**Reply by fahadsohail482** [reply]
> I still face problem when trying to reset() data. Please help

[Link](https://codesandbox.io/s/zealous-sound-bwmud?file=/pages/index.js)
